day2/day2.erl

```diff
@@ -0,0 +1,59 @@
+-module(day2).
+-export([test_new_computer/0, find_input_pair_checksum/0]).
+
+read_input(FileName) ->
+    {ok, Data} = file:read_file(FileName),
+    binary:split(Data, [<<",">>], [global]).
+
+process_input(L) ->
+    [ list_to_integer(binary_to_list(X)) || X <- lists:append(lists:sublist(L, length(L) - 1), [binary:replace(lists:nth(length(L), L), <<"\n">>, <<"">>)]) ].
+
+replace_value(L, I, X) ->
+    Head = lists:sublist(L, 1, I - 1),
+    Tail = lists:sublist(L, I + 1, length(L)),
+    Head ++ [X] ++ Tail.
+
+change_noun_and_verb(X, Y) ->
+    Input = process_input(read_input("day2.input")),
+    InputReplaceOne = replace_value(Input, 1 + 1, X),
+    replace_value(InputReplaceOne, 2 + 1, Y).
+
+execute_intcode(L) ->
+    execute_intcode(L, 1).
+
+execute_intcode(L, I) ->
+    Opcode = lists:sublist(L, I, 4),
```

Comment:
Ignoring sublist usage but, here is another opportunity to pattern match!

```erlang
execute_intcode(L, I) ->
    case lists:sublist(L, I, 4) of
        [1, Pos1, Pos2, PosReplace] -> do_sum();
        [2, Pos1, Pos2, PosReplace] -> do_mult();
        [99 | _] -> L
    end.
```

This also just lets the code crash if an invalid opcode is encountered. Higher level code can handle the exception if necessary. This is called the happy case and we should embrace coding in it! :)